src/mapgen/mapgen.cpp

```diff
@@ -468,6 +468,8 @@ void Mapgen::lightSpread(VoxelArea &a, std::queue<std::pair<v3s16, u8>> &queue,
 void Mapgen::calcLighting(v3s16 nmin, v3s16 nmax, v3s16 full_nmin, v3s16 full_nmax,
 	bool propagate_shadow)
 {
+	if (g_settings->getBool("enable_dynamic_shadows"))
+		propagate_shadow = false;
```

Comment:
This looks like a problem.

The mapgen runs on the server, it must not change its results based on purely visual settings set on the client (never mind the fact that this can't work with _multiple_ clients).

Reply:
Ok, we can remove it without a problem. It's just we'll get double shadows (ShadowMap and propagated one) in remote servers and in local ones.

Reply:
Done.

Reply:
Shouldn't mesh generation be able to remove the shadows encoded in the map?

Reply:
I don't know. I'm going to take a look on that.

Reply:
Added to future improvements.

Reply:
Hi again, I tried to fix this, but as far as I know the static shadow can't be removed in client or at least I don't know how to do it.